### PR TITLE
Feature/tier two testing

### PR DESCRIPTION
**Pull request by philngo**
A stab at testing and auditing production data. (WIP)

- Adds an auditing app
- Adds tasks and command for running audits
- Tracks audits run and stores failures in table `auditing_auditresult`

(Based off of feature/time-based-derivatives, which serves a dual testing purpose.)

### CODE REVIEW

File: auditing/tests/test_trace_should_have_meter_run.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        project = ProjectFactory.create()
+        trace1 = TraceFactory.create()
+
+        meter_run = MeterRunFactory.create(
+            trace=trace1,
+            project=project
+        )
+
+        # no result before audit is run
```
Comment (reetu) [inline auditing/tests/test_trace_should_have_meter_run.py]: Each of these sections (no result, no failure, failure) deserves to be in its own test method, as they each verify a different aspect of functionality.

File: auditing/tests/test_trace_should_have_project.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+class TraceShouldHaveProjectTest(TestCase):
+
+    def test_task(self):
+
+        # no result before audit is run
+        audit_results = AuditResult.objects.all()
+        assert len(audit_results) == 0
+
+        # no traces
```
Comment (reetu) [inline auditing/tests/test_trace_should_have_project.py]: @philngo Each of these sections deserves to be in its own test method too.

File: auditing/tests/test_trace_with_meter_result_should_have_baseline_model_result.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            project=project
+        )
+
+        meter_result1 = MeterResultFactory.create(
+            trace=trace1,
+            project=project,
+            meter_run=meter_run1
+        )
+
+        # no result before audit is run
```
Comment (reetu) [inline auditing/tests/test_trace_with_meter_result_should_have_baseline_model_result.py]: @philngo Each of these sections deserves to be in its own test method too.

File: auditing/tests/test_trace_with_meter_run_should_have_meter_result.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+class TraceWithMeterRunShouldHaveMeterResultTest(TestCase):
+
+    def test_task(self):
+
+        project = ProjectFactory.create()
+        trace1 = TraceFactory.create()
+        trace2 = TraceFactory.create()
+
+        # no result before audit is run
```
Comment (reetu) [inline auditing/tests/test_trace_with_meter_run_should_have_meter_result.py]: @philngo Each of these sections deserves to be in its own test method too.

File: auditing/tests/test_trace_should_have_meter_run.py (hunk 1 above)
Comment (philngo) [inline auditing/tests/test_trace_should_have_meter_run.py]: You are probably right about that, @reetu. It's not too difficult to make the change. I will probably reorganize this test (and others) to use the setUp method. Thanks for reviewing.

**Pull request merged**